src/plugins/resolver/mutex.c

```diff
@@ -3,5 +3,8 @@
 #include <pthread.h>
 
 // every resolver should use the same mutex
-pthread_mutex_t elektra_resolver_mutex =
-	PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
```

Comment:
So we simply need an ifdef to either use PTHREAD_MUTEX_RECURSIVE or PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP